sstables/mx/reader.cc

```diff
@@ -403,9 +402,7 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
                 goto partition_start_label;
             } else if (_flags.is_range_tombstone()) {
                 _state = state::RANGE_TOMBSTONE_MARKER;
-                goto range_tombstone_marker_label;
             // case state::RANGE_TOMBSTONE_MARKER:
-        range_tombstone_marker_label:
```

Comment:
This patch does a lot of things (like here removing the gotos and labels which I wondered in the previous patch why you didn't remove them) that have nothing to do with the patch's description.

Reply:
True, I will separate them in the final PR